src/services/whatsapp.service.ts

```diff
@@ -24,52 +27,57 @@ export class WhatsappService {
     try {
       const phoneF = `521${phone}@s.whatsapp.net`;
 
-      const response = await axios.post(
-        `${this.apiUrl}/messages/text`,
-        {
-          to: phoneF,
-          channel: this.channelId,
-          body: message,
-        },
-        {
-          headers: this.getHeaders(),
-        },
+      const response = await firstValueFrom(
+        this.httpService.post(
+          `${this.apiUrl}/messages/text`,
+          {
+            to: phoneF,
+            channel: this.channelId,
+            body: message,
+          },
+          {
+            headers: this.getHeaders(),
+          },
+        ),
       );
 
       return response.data;
     } catch (error) {
+      this.logger.error(`Error al enviar el mensaje: ${error.message}`);
       this.handleError(error);
     }
   }
 
-  async sendInteractiveMessage(phone: string): Promise<any> {
+  async sendInteractiveMessage(phone: string, message: string): Promise<any> {
     try {
       const formattedPhone = `521${phone}@s.whatsapp.net`;
       const data = {
         to: formattedPhone,
         channel: this.channelId,
         type: 'button',
         body: {
-          text: 'empty',
+          text: message,
         },
         action: {
           buttons: [
             {
-              id: 'id',
+              id: 'confirm',
```

Comment:
💭 id `confirm`? 🤔   is there any documentation that tells what `id` should be? doesn't seem like a very well identifier